src/sisyphus/core.clj

```diff
@@ -173,13 +174,19 @@
   (edn/read-string
    (slurp path)))
 
+(def parse-options
+  [["-w" "--workflow WORKFLOW" "which workflow this worker is assigned to"]])
+
 (defn -main
   [& args]
   (try
     (log/info! "sisyphus worker rises:" log/gce-instance-name)
-    (let [options (:options (cli/parse-opts args rabbit/parse-options))
+    (let [options (:options (cli/parse-opts args parse-options))
+          rabbit-options
+          (when-let [workflow (:workflow options)]
+            (rabbit/rabbit-metadata workflow))
```

Comment:
Doesn't `when-let` return `nil` in the nil case? Will that work here?

Reply:
Yes, it is merged in with all the other sources of this info above. `(merge m nil) --> m` so that's exactly what we want.